### DISCUSSION TITLE
Discord.js bot not playing audio

### DISCUSSION THREAD

**Discussion by SamDizzle** [discordjs/discord.js#9714] [Q&A] [answered]
Hey everyone,

i just created my first discord music bot (or did my best attempt since i'm very new to discord.js or javascript in general.)

Everything seems to be working, i think i've updated everything to work with discord.js v14 but the bot it doesn't start playing audio when coming into the voice channel. I get no errors or anything.

It joins, shows the embed of the current song playing but doesn't start playing audio.

Also the buttons don't work as they just say that the interaction failed. But honestly i can live with that. I just want the audio to start playing lol.

Here is a link to my updated code without YouTube functionality since it's against TOS: https://pastebin.com/DQj2Y2n4

There's a high probability that i did something wrong or forgot something, but i have no clue what.

Any help would be greatly appreciated!

**Comment by sdanialraza** [comment]
You should have a solid understanding of JavaScript before you start with discord.js, as for your code, what you're doing seems to be against YouTube TOS and by proxy Discord Developer TOS so I don't think you'd get much support with that. As for other things, I'd recommend joining the [discord.js Discord server](discord.gg/djs) and asking there for a faster response.

**Reply by SamDizzle** [reply]
Oh really? I didn't know that. I'll just remove the youtube part from my code then. Spotify and soundcloud is fine right?

**Reply by sdanialraza** [reply]
Yep, and I am not sure about Spotify and Soundcloud, so you'd need to look into that by yourself

**Reply by SamDizzle** [reply]
Will do, thanks for letting me know though!

**Reply by sdanialraza** [reply]
Of course

**Reply by SamDizzle** [reply]
I removed the youtube part from my code, i also asked in the discord.js discord server, but i'm not getting any responses. Would you be willing to look over my code and see what's wrong?

**Reply by sdanialraza** [reply]
Not a lot of people are experienced as far as @discordjs/voice goes so you might have to wait before someone responds. Unfortunately, I haven't worked with it a lot either, but don't worry, I am sure someone will help you eventually.

**Reply by SamDizzle** [reply]
Oh okay, well i've updated my post here as well without the youtube side of things. Guess i'll just have to wait like you said. Thanks for your help either way.

**Reply by sdanialraza** [reply]
You're welcome, it'd be better if you just asked there as you'd be asking the same thing in 2 places and I am not sure if many people check discussions compared to the Discord server

**Comment by Jiralite** [comment, marked as answer]
Resolved in Discord.[^1]

[^1]: https://discord.com/channels/222078108977594368/1130903727839121408

**Reply by MiniProject25** [reply]
and it's locked now. what's the point even?

**Reply by Amgelo563** [reply]
The point is that the thread answers the user's question and can help others that are also trying to play from Spotify. It's not for unknown users to hijack the thread and reply with their own issues, so it being locked should be irrelevant. Also this is a 3 year old post, if you have an issue and the thread didn't answer it, you should make a new one
